packit_service/worker/handlers/bodhi.py

```diff
@@ -0,0 +1,130 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+"""
+This file defines classes for job handlers related to Bodhi
+"""
+import logging
+from typing import Optional
+
+from packit.constants import DEFAULT_BODHI_NOTE
+
+from packit.exceptions import PackitException
+
+from packit.api import PackitAPI
+from packit.config.aliases import get_branches
+
+from packit.config import JobConfig, JobType, PackageConfig
+from packit.local_project import LocalProject
+from packit.utils.repo import RepositoryCache
+from packit_service.constants import CONTACTS_URL, KojiBuildState
+from packit_service.worker.events.koji import KojiBuildEvent
+from packit_service.worker.handlers.abstract import (
+    JobHandler,
+    TaskName,
+    configured_as,
+    reacts_to,
+)
+from packit_service.worker.result import TaskResults
+
+logger = logging.getLogger(__name__)
+
+
+@configured_as(job_type=JobType.bodhi_update)
+@reacts_to(event=KojiBuildEvent)
+class CreateBodhiUpdateHandler(JobHandler):
+    """
+    This handler can create a bodhi update for successful Koji builds.
+    """
+
+    topic = "org.fedoraproject.prod.buildsys.build.state.change"
+    task_name = TaskName.bodhi_update
+
+    def __init__(
+        self,
+        package_config: PackageConfig,
+        job_config: JobConfig,
+        event: dict,
+    ):
+        super().__init__(
+            package_config=package_config,
+            job_config=job_config,
+            event=event,
+        )
+
+        # lazy properties
+        self._koji_build_event: Optional[KojiBuildEvent] = None
+
+    @property
+    def koji_build_event(self):
+        if not self._koji_build_event:
+            self._koji_build_event = KojiBuildEvent.from_event_dict(
+                self.data.event_dict
+            )
+        return self._koji_build_event
+
+    def pre_check(self) -> bool:
+        """
+        We react only on finished builds (=KojiBuildState.open state)
+        and configured branches.
+        :return:
+        """
+        if self.koji_build_event.state != KojiBuildState.open:
+            logger.debug(
+                f"Skipping build '{self.koji_build_event.build_id}' "
+                f"on '{self.koji_build_event.git_ref}'. "
+                f"Build not finished yet."
+            )
+            return False
+
+        if self.koji_build_event.git_ref not in (
+            configured_branches := get_branches(
+                # TODO: pick a better default since for rawhide,
+                #   update is created automatically
+                *self.job_config.metadata.dist_git_branches,
+                default="rawhide",
+                default_dg_branch="rawhide",
```

Comment:
So, if someone omits `dist_git_branches` for `bodhi_update` job, then this handler will try to create an update for the rawhide build as well?
Then setting [`default=None` should make `get_branches()` return empty set](https://github.com/packit/packit/blob/dce0380bd05ad6df09f14a7ca0a23bb36a7d9932/packit/config/aliases.py#L152)?

```suggestion
                *self.job_config.metadata.dist_git_branches,
                # Don't use "rawhide" as default since update for rawhide is created automatically
                default=None,
```

Reply:
I was more thinking about using `fedora-stable` alias as a default. What do you think?

Reply:
+1 fedora-stable or require the list the be set

Reply:
I'd vote for "require the list the be set", but `fedora-stable` is fine as well if it's explicitly stated in the documentation.

Reply:
I'll use `fedora-stable` and document it.